variables.mk

```diff
@@ -62,11 +97,14 @@ REBAR_FIRRTL_DIR = $(base_dir)/tools/firrtl
 #########################################################################################
 # names of various files needed to compile and run things
 #########################################################################################
-long_name = $(PROJECT).$(MODEL).$(CONFIG)
+long_name = $(MODEL_PACKAGE).$(MODEL).$(CONFIG)
 
-# if building from rocketchip, override the long_name to match what they expect
-ifeq ($(SBT_PROJECT),rebarrocketchip)
-	long_name=$(PROJECT).$(CONFIG)
+# match the long_name to what the specific generator will output
+ifeq ($(GENERATOR_PACKAGE),freechips.rocketchip.system)
+	long_name=$(CONFIG_PACKAGE).$(CONFIG)
+endif
+ifeq ($(GENERATOR_PACKAGE),hwacha)
+	long_name=$(MODEL_PACKAGE).$(CONFIG)
```

Comment:
Wait does this work? I thought hwacha's generator mirrored what rocket-chip does...

Reply:
I think this does the equivalent thing as rocket-chip, just the naming is different. @abejgonzalez has the reasoning for why this appears different.

Reply:
This works. However, the `hwacha` generator uses the following: `names.topModuleProject + "." + names.configs` whereas `rocket-chip` does the following: `names.configProject + "." + names.configs`. So they are slightly different

Reply:
This change reflects that